### PR TITLE
GitHub issues directory structure

### PR DESCRIPTION
**Pull request by thefishua**
Bare bones directory structure for the issues section for the DYNO project

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 build:
 	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/main cmd/*.go
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/issue cmd/internal/lambda-issue/*.go
```
Comment (The-TT-Hacker) [inline Makefile]: I think we can change this to
`CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/ ./cmd/...`

File: cmd/internal/lambda-issue/main.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package main
```
Comment (The-TT-Hacker) [inline cmd/internal/lambda-issue/main.go]: dont need the `internal` directory. just `cmd/issues-lambda`

File: Makefile [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 build:
 	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/main cmd/*.go
-
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/ ./cmd/...
```
Comment (The-TT-Hacker) [inline Makefile]: delete line 4, `CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/ ./cmd/...` will compille everything in cmd

**Comment by The-TT-Hacker** [comment]
Maybe think about an including an interface for the different targets e.g. github, gitlab, jira, gitea etc.

**Comment by thefishua** [comment]
> Maybe think about an including an interface for the different targets e.g. github, gitlab, jira, gitea etc.

Ill probably make another branch for this as Im still fixing other issues with the code I have written

**Comment by The-TT-Hacker** [comment]
the end to end tests and terraform need the api code compiled into /bin/main. might need to do something to fix that or maybe just leave to old makefile command in for now

**Comment by thefishua** [comment]
> the end to end tests and terraform need the api code compiled into /bin/main. might need to do something to fix that or maybe just leave to old makefile command in for now

Is the new cmd binary just the same as the main file or does it include the code I have written in lambda-issue ?
@The-TT-Hacker

**Comment by The-TT-Hacker** [comment]
> > the end to end tests and terraform need the api code compiled into /bin/main. might need to do something to fix that or maybe just leave to old makefile command in for now
> 
> Is the new cmd binary just the same as the main file or does it include the code I have written in lambda-issue ? @The-TT-Hacker

yeah but it needs to be called main instead of cmd

File: internal/github/github_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	"context"
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"golambda/internal/github"
+	"os"
+	"testing"
+)
+
+func getEnv() string {
+	err := godotenv.Load("../../.env")
```
Comment (The-TT-Hacker) [inline internal/github/github_test.go]: dont load from env files in unit tests

File: internal/github/github_test.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	client := github.CreateClient(ctx, &token)
+	actualRepo := github.GetRepo(ctx, client, "thefishua", "Demo-Server")
+	assert.Equal(t, "thefishua/Demo-Server", *actualRepo.FullName)
+}
+
+func TestInvalidGetRepo(t *testing.T) {
+	token := getEnv()
+	ctx := context.Background()
+	client := github.CreateClient(ctx, &token)
+	actualRepo := github.GetRepo(ctx, client, "thefishua", "Demo-Server")
```
Comment (The-TT-Hacker) [inline internal/github/github_test.go]: no need to test the github client and dont make network calls in unit tests

File: Makefile [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 build:
 	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/api/main ./cmd/api/...
-
+
```
Comment (The-TT-Hacker) [inline Makefile]: remove tab here

**Review by The-TT-Hacker — APPROVED** [review]

**Pull request merged**